nUpdate.Shared/Updating/UpdateManager.cs

```diff
@@ -472,5 +486,56 @@ bool Validate(KeyValuePair<UpdateVersion, string> filePathItem)
             Cleanup();
             return false;
         }
+
+        /// <summary>
+        ///     Convert <paramref name="updatePackageUri"/> (its parent directory <paramref name="literalVersion"/>) to the one that
+        ///     corresponds to the directory where aslo the configuration file that is located in <see cref="UpdateManager.UpdateConfigurationFileUri"/>.
+        ///     If <see cref="UseDynamicUpdateUri"/> is false, return the original value.
+        /// </summary>
+        /// <param name="updatePackageUri">The update-packages uri. <see cref="UpdateConfiguration.UpdatePackageUri"/></param>
+        /// <param name="literalVersion">The update-packages literalVersion. <see cref="UpdateConfiguration.LiteralVersion"/></param>
+        /// <returns>
+        ///     Return converted URI.
+        ///     If <see cref="UpdateManager.UseDynamicUpdateUri"/> is false, return the original value.
+        /// </returns>
+        /// <exception cref="System.ArgumentNullException">updatePackageUri or literalVersion is null.</exception>
+        /// <exception cref="System.ArgumentException">literalVersion is a valid value. Reference <see cref="UpdateVersion"/></exception>
+        /// <exception cref="System.InvalidOperationException">updatePackageUri.AbsoluteUri is a valid uri.</exception>
+        private Uri ConvertPackageUri(Uri updatePackageUri, string literalVersion)
+        {
+            if (!UseDynamicUpdateUri)
+                return updatePackageUri;
+            if (updatePackageUri == null)
+                throw new ArgumentNullException(nameof(updatePackageUri));
+            if (literalVersion is null)
+                throw new ArgumentNullException(nameof(literalVersion));
+            if (!UpdateVersion.IsValid(literalVersion))
+                throw new ArgumentException($"{literalVersion} is not valid value.", nameof(literalVersion));
```

Comment:
You replace the string argument by the name of the parameter. It would be better to use its value instead.

Reply:
Replaced